File: solutions/jose-rivas/csharp/p-02-arrays/exercise-01.cs
using System;

class Program
{
    static void Main()
    {
        try
        {
            Console.Write("Entrez le nombre d'elements : ");
            int N = int.Parse(Console.ReadLine());

            if (N < 1)
            {
                Console.WriteLine("Erreur : N doit etre >= 1");
                return;
            }

            int[] tableau = new int[N];

            Console.WriteLine("Entrez les nombres :");

            for (int i = 0; i < N; i++)
            {
                tableau[i] = int.Parse(Console.ReadLine());
            }

            int max = tableau[0];

            for (int i = 1; i < N; i++)
            {
                if (tableau[i] > max)
                {
                    max = tableau[i];
                }
            }

            Console.WriteLine("Le maximum est : " + max);
        }
        catch
        {
            Console.WriteLine("Erreur : veuillez entrer uniquement des nombres entiers.");
        }
    }
}

File: solutions/jose-rivas/java/p-02-arrays/exercise-01.java
import java.util.Scanner;

public class MaximumElement {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        try {
            System.out.print("Entrez le nombre d'elements : ");
            int N = scanner.nextInt();

            if (N < 1) {
                System.out.println("Erreur : N doit etre >= 1");
                return;
            }

            int[] tableau = new int[N];

            System.out.println("Entrez les nombres :");

            for (int i = 0; i < N; i++) {
                tableau[i] = scanner.nextInt();
            }

            int max = tableau[0];

            for (int i = 1; i < N; i++) {
                if (tableau[i] > max) {
                    max = tableau[i];
                }
            }

            System.out.println("Le maximum est : " + max);

        } catch (Exception e) {
            System.out.println("Erreur : veuillez entrer uniquement des nombres entiers.");
        }

        scanner.close();
    }
}